### DISCUSSION TITLE
login page for http basic auth

### DISCUSSION THREAD

**Discussion by jeffective** [zauberzeug/nicegui#1629] [Q&A] [answered]
### Question

I am trying to implement a login page for http basic auth.

After the client has input their username and password and pressed a login button, how can i cause the client to be redirected to a new page with the appropriate http basic auth headers?

ui.open does not allow adding additional headers

**Comment by rodja** [comment, marked as answer]
You do not need any NiceGUI features for Basic Auth. You should be able to follow https://fastapi.tiangolo.com/advanced/security/http-basic-auth/. No need to redirect with `ui.open`.

**Reply by ghost** [reply]
@rodja I am still very confusing the Auth login page, I want to use FastAPI-Users because the full functions, everything is OK except when I try to implement a Login UI page,   FastAPI-Users provide the auth route like `/auth/jwt/login` and follow the normal FastAPI endpoint call, but my Login UI page is seperated with it, my thinking is to click the Login button to trigger the `/auth/jwt/login` endpoint, but it seems wrong, the problems are: 1) I don't know how to trigger the `/auth/jwt/login` endpoint, 2) if I use requests or fastapi.testclient to execute the endpoint it return another session.

the code snippet is:
```py
#  fastapi_users generate the /login and /logout route
app.include_router(
    fastapi_users.get_auth_router(auth_backend),
    prefix="/auth/jwt",
    tags=["auth"],
)

# dependency callable to retrieve currently authenticated user
current_active_user = fastapi_users.current_user(active=True)

# the Login UI for Login activity
@ui.page('/login')
def login() -> Optional[RedirectResponse]:
    def try_login() -> None:  
        # ???
        # how to `POST /auth/jwt/login' and then from the response status code to redirect to the working page
        # ???
        # below codes can't work correctly, the current_active_user is not authorized:
        user_data = {"grant_type": "password", "username": email.value, "password": password.value}
        head = {"Content-Type": "application/x-www-form-urlencoded; charset=UTF-8", 'Connection': 'close'}
        fastapi_client = TestClient(app)
        response = fastapi_client.post('http://internal-api/auth/jwt/login', data=user_data, headers=head)
        if response.status_code == 200:
            RedirectResponse('/')
        else:
            ui.notify('Wrong username or password', color='negative')
            RedirectResponse('/login')

    with ui.card().classes('absolute-center'):
        email = ui.input('Email').on('keydown.enter', try_login)
        password = ui.input('Password', password=True, password_toggle_button=True).on('keydown.enter', try_login)
        ui.button('Log in', on_click=try_login)
    return None

# will response '401 Unauthorized' if the user is unauthorized
@ui.page('/', dependencies=[Depends(current_active_user )])
def main_page() -> None:
    with ui.column().classes('absolute-center items-center'):
        ui.label(f'Hello World')
```

Many thanks for your advice!

**Reply by rodja** [reply]
I have only glanced over your code, but maybe you could call the method `fastapi_users.get_auth_router(auth_backend)` directly instead of requesting it via http...
Still, this has nothing to do with http basic auth so I would kindly ask you to start a new Q&A to keep things orderly.
